packages/ckeditor5-mention/src/mention.ts

```diff
@@ -35,7 +35,28 @@ export default class Mention extends Plugin {
 	 * @param viewElement
 	 * @param data Additional data to be stored in the mention attribute.
 	 */
-	public toMentionAttribute( viewElement: Element, data?: MentionAttribute ): MentionAttribute | undefined {
+	public toMentionAttribute<MentionData extends object>(
```

Comment:
Consider using `{ [ key: string ]: unknown }` like in `packages/ckeditor5-mention/src/mentioncommand.ts`) instead of `object`.

[TypeScript Playground](https://www.typescriptlang.org/play?#code/DYUwLgBAhgXBDeEDaBrEBPOBnMAnAlgHYDmAunAK6EqED2A7oRAL4QC8CzA3ALABQUAHQAzWrV59+oSACM4tGQCsQAY0gd43fjJFiJU8BBVwASqtq4AJgB4cBEgBoIVGg0IA+dpwkrd4oA).

This also applies to the line 59 in this file and lines 88 and 103 in `packages/ckeditor5-mention/src/mentionediting.ts`.

Reply:
Why? The data is just consumed, not produced.

Reply:
It's always better to be explicit about what type you expect. The `object` type means **any** object, including `() => {}` and `[]`, since both are objects in JavaScript. See the [TypeScript Playground](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABHARgKwKbQCoE8AOGAFKmgFzLpZQCUFAbnDACaIDeAvgLABQvp1PISKcaAbkSSpUgPQzEAQQDOiDAA9C0DM35UcBYkRqIAvAD52HcbPmBeDcClO7sz7hAbQC616dLmIHvXlBIWAREACcsODDmIWJSCgAlSOiAHiUoMJgwAHMAGkRwAGswOAB3MDM6REYWSwCeCIgomIMRKwlvG0UVdU0obV5G5tiiI1MLUQ7EXwB1AEMwsCy8xDmVUoxVDWpB5Ja3TynO2YWlnPy1xA2tvvqg6HgkARcMGZgoAAsAZQzlgGkMLglCR0BQ2IhXIhCoCKOlMjlEO4KEUSuVEFYGExWJx6s8oLE3p8fvDsgCgW0vJ1pvJlDdqAMeHiCe9vr8cmTgWNzJZKSdFssLutNr0doy9PiDITWSSOUQPJTjvJ5vzzqshXToEA) example.

In this case, to be more explicit and only allow objects with string keys, as we [show in the documentation](https://ckeditor.com/docs/ckeditor5/latest/features/mentions.html#customizing-the-output), we should use `{ [ key: string ]: unknown }` or `Record<string, unknown>` (I'm a fan of `Record` because it's more concise, but in this case it doesn't matter).